src/taipy/core/config/data_node_config.py

```diff
@@ -310,222 +351,232 @@ def _configure(
+            cls._STORAGE_TYPE_VALUE_CSV: cls._configure_csv,
+            cls._STORAGE_TYPE_VALUE_EXCEL: cls._configure_excel,
+            cls._STORAGE_TYPE_VALUE_IN_MEMORY: cls._configure_in_memory,
+            cls._STORAGE_TYPE_VALUE_GENERIC: cls._configure_generic,
+            cls._STORAGE_TYPE_VALUE_JSON: cls._configure_json,
+            cls._STORAGE_TYPE_VALUE_PARQUET: cls._configure_parquet,
+        }
+
+        if storage_type in cls._ALL_STORAGE_TYPES:
+            return configuration_map[storage_type](id=id, scope=scope, **properties)
+
+        return cls.__configure(id, storage_type, scope, **properties)
 
-    @staticmethod
+    @classmethod
     def _configure_csv(
+        cls,
         id: str,
-        default_path: str = None,
-        has_header: bool = True,
-        exposed_type=_DEFAULT_EXPOSED_TYPE,
-        scope=_DEFAULT_SCOPE,
+        default_path: Optional[str] = None,
+        has_header: Optional[bool] = None,
+        exposed_type: Optional[str] = None,
+        scope: Optional[Scope] = None,
         **properties,
     ):
         """Configure a new CSV data node configuration.
 
         Parameters:
             id (str): The unique identifier of the new CSV data node configuration.
-            default_path (str): The default path of the CSV file.
-            has_header (bool): If True, indicates that the CSV file has a header.
-            exposed_type: The exposed type of the data read from CSV file. The default value is `pandas`.
-            scope (Scope^): The scope of the CSV data node configuration. The default value
+            default_path (Optional[str]): The default path of the CSV file.
+            has_header (Optional[bool]): If True, indicates that the CSV file has a header.
+            exposed_type (Optional[str]): The exposed type of the data read from CSV file.
+                The default value is `pandas`.
+            scope (Optional[Scope^]): The scope of the CSV data node configuration. The default value
                 is `Scope.SCENARIO`.
             **properties (Dict[str, Any]): A keyworded variable length list of additional
                 arguments.
         Returns:
             `DataNodeConfig^`: The new CSV data node configuration.
         """
-        section = DataNodeConfig(
-            id,
-            DataNodeConfig._STORAGE_TYPE_VALUE_CSV,
-            scope=scope,
-            default_path=default_path,
-            has_header=has_header,
-            exposed_type=exposed_type,
-            **properties,
-        )
-        Config._register(section)
-        return Config.sections[DataNodeConfig.name][id]
+        if default_path is not None:
+            properties[cls._OPTIONAL_DEFAULT_PATH_CSV_PROPERTY] = default_path
+        if has_header is not None:
+            properties[cls._OPTIONAL_HAS_HEADER_CSV_PROPERTY] = has_header
+        if exposed_type is not None:
+            properties[cls._OPTIONAL_EXPOSED_TYPE_CSV_PROPERTY] = exposed_type
```

Comment:
This section is to ignore all `None` properties.

However, the code is really ugly atm. If anyone know how to refactor this, please let me know to make it easier to maintain.

Reply:
I believe that s fine!